### PR TITLE
Add chat functionality

### PR DESCRIPTION
**Pull request by Vixmit**

### CODE REVIEW

File: Angular/src/app/messenger-page/right-box/sender-bar/sender-bar.component.ts [hunk 1]
```diff
@@ -1,15 +1,32 @@
...
-  constructor() { }
+  private conversation: ConversationModel;
 
   ngOnInit() {
+    this.chatService.currentConversation
+    .subscribe((data) => this.conversation = data);
   }
 
+  sendMessage(messageInput: any) {
+    if (messageInput.value === '') { // TODO do not allow whitespace strings - regex?
```
Comment (PBlajer) [inline Angular/src/app/messenger-page/right-box/sender-bar/sender-bar.component.ts]: first trim and then compare to empty string
https://www.w3schools.com/jsref/jsref_trim_string.asp

File: Angular/src/app/messenger-page/left-box/nav-bar/send-message-modal/send-message-modal.component.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+                    </div>
+                    <div class="form-group">
+                        <label for="message-text" class="col-form-label">Message:</label>
+                        <textarea class="form-control" id="message-text"></textarea>
+                    </div>
+                </form>
+            </div>
+            <div class="modal-footer">
+                <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
+                <button (click)="sendNewTextMessage()" type="button" class="btn btn-primary">Send message</button>
```
Comment (PBlajer) [inline Angular/src/app/messenger-page/left-box/nav-bar/send-message-modal/send-message-modal.component.html]: Does this work? .ts file for this component is empty. 
Now that we can send messages inside dialog-box component and create conversations by adding friends, maybe this component isn't needed anymore?

File: ASP.NET API/WAVC_WebApi/Controllers/FriendRequestsController.cs [hunk 3]
```diff
@@ -22,15 +21,23 @@ public class FriendRequestsController : ControllerBase
         private readonly ApplicationDbContext _dbContext;
         private readonly UserManager<ApplicationUser> _userManager;
         private readonly FriendsManager _friendsManager;
-        private readonly IHubContext<FriendRequestHub, IFriendRequestClient> _friendRequestHubContext;
+        private readonly IHubContext<FriendRequestHub, IFriendRequestClient> _friendRequestFriendRequestHubContext;
```
[the remark this answers is not in the document]
Comment (Vixmit) [reply ASP.NET API/WAVC_WebApi/Controllers/FriendRequestsController.cs]: O.o Sorry Not intended. I'm wondering how it happened.. Autocompleter probably...

File: Angular/src/app/messenger-page/right-box/sender-bar/sender-bar.component.ts (hunk 1 above)
Comment (Vixmit) [reply Angular/src/app/messenger-page/right-box/sender-bar/sender-bar.component.ts]: Good catch :D  I was searching for any function called strip (python), something like that,  and it's apparently called differently in js.

File: Angular/src/app/messenger-page/left-box/nav-bar/send-message-modal/send-message-modal.component.html (hunk 2 above)
Comment (Vixmit) [reply Angular/src/app/messenger-page/left-box/nav-bar/send-message-modal/send-message-modal.component.html]: This modal will be ready used for creating group conversations. Recipients separated by semicolon or sth like that. Not yet implemented

File: ASP.NET API/WAVC_WebApi/Hubs/MessagesHub.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            {
+                await Groups.AddToGroupAsync(Context.ConnectionId, id.ToString());
+            }
+
+            await base.OnConnectedAsync();
+        }
+
+        public async Task JoinConversation(int conversationId)
+        {
+            await Groups.AddToGroupAsync(Context.ConnectionId, conversationId.ToString());
```
[the remark this answers is not in the document]
Comment (Vixmit) [reply ASP.NET API/WAVC_WebApi/Hubs/MessagesHub.cs]: Yeah you're right. This function was written at 3am and was intended just to work.

**Pull request merged**